bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragmentReplicator.java

```diff
@@ -319,7 +321,7 @@ public void readComplete(int rc, LedgerHandle lh,
                 for (BookieSocketAddress newBookie : newBookies) {
                     bkc.getBookieClient().addEntry(newBookie, lh.getId(),
                             lh.getLedgerKey(), entryId, toSend.retainedSlice(),
-                            multiWriteCallback, dataLength, BookieProtocol.FLAG_RECOVERY_ADD);
+                            multiWriteCallback, dataLength, BookieProtocol.FLAG_RECOVERY_ADD, ledgerType);
```

Comment:
You're never doing a force for fragment replication, so I'd make this add a forcing write, regardless of ledger type.

Reply:
with FLAG_RECOVERY_ADD on the bookie we will have ackAfterForce=true